src/Lecture3.hs

```diff
@@ -210,11 +237,24 @@ types that can have such an instance.
   implement instances! No spoilers :)
 -}
 
--- instance Foldable Weekday where
--- instance Foldable Gold where
--- instance Foldable Reward where
--- instance Foldable List1 where
--- instance Foldable Treasure where
+instance Foldable List1 where
+  foldMap :: Monoid m => (a -> m) -> List1 a -> m
+  foldMap fm (List1 x0 []) = fm x0
+  foldMap fm (List1 x0 (x1:xs)) = fm x0 <> foldMap fm (List1 x1 xs)
```

Comment:
That's already the default implementation of `foldMap` so there's not much benefit in implementing it explicitly in exactly the same way 🙂 
But maybe you can think of a different implementation 🤔